test/bench.cpp

```diff
@@ -1,90 +1,392 @@
 #include <lorina/bench.hpp>
-#include <cstdio>
+#include <lorina/lorina.hpp>
 #include <fstream>
 #include <string>
 #include <vector>
+#include <stdio.h>
+
+#include <SDL.h>
+#include <SDL_ttf.h>
+
+
+// Define screen dimensions
+#define SCREEN_WIDTH    800
+#define SCREEN_HEIGHT   600
+
+
+#define FONT_PATH   "../assets/ttf/DejaVuSansMono.ttf"
 
 using namespace lorina;
 
-struct bench_statistics
-{
-  uint32_t number_of_inputs = 0;
-  uint32_t number_of_outputs = 0;
-  uint32_t number_of_dffs = 0;
+struct parametrs {
+    std::vector<std::string> inputs;
+    std::string type = NULL;
+    std::string output = NULL;
+
+    //coordinates of first and second points
+    float fp_x= 0;
+    float fp_y = 0;
+    float sp_x = 0;
+    float sp_y = 0;
+
+    int coeff_x = 0;
+    int coeff_y = 0;
 
-  /* lines without input and outputs */
-  uint32_t number_of_lines = 0;
+
+    parametrs(const std::vector<std::string> &inputs, const std::string &type, const std::string &output)
+            : inputs(inputs), type(type), output(output) {}
+};
+
+struct bench_statistics {
+    uint32_t number_of_inputs = 0;
+    uint32_t number_of_outputs = 0;
+    uint32_t number_of_dffs = 0;
+
+    /* lines without input and outputs */
+    uint32_t number_of_lines = 0;
+    uint32_t number_of_edges = 0;
+
+    std::vector<parametrs> element;
 };
 
-class bench_statistics_reader : public bench_reader
-{
+
+void addEdge(bench_statistics &stats, const std::vector<std::string> &inputs, const std::string &type = "",
+             const std::string &output = ""){
+    stats.element.emplace_back(inputs, type, output);
+}
+
+void sortGraph(bench_statistics &stats, std::vector<std::vector<parametrs>> &matrix) {
+    std::vector<parametrs> temp;
+
+    for (const auto &iter : stats.element) {
+        std::string type = iter.type;
```

Comment:
Avoid unnecessary copying
```suggestion
        const std::string &type = iter.type;
```